src/Persistence/Yaml/NamedObjectTypeInspector.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using YamlDotNet.Serialization;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.Yaml;
+
+internal class NamedObjectTypeInspector : ITypeInspector
+{
+    private readonly ITypeInspector _innerTypeInspector;
+
+    public NamedObjectTypeInspector(ITypeInspector innerTypeInspector)
+    {
+        _innerTypeInspector = innerTypeInspector;
+    }
+
+    public IEnumerable<IPropertyDescriptor> GetProperties(Type type, object? container)
+    {
+        var properties = _innerTypeInspector.GetProperties(type, container);
```

Comment:
not for this PR, but we should see if caching properties brings some benefits

Reply:
_innerTypeInspector is actually is caching it already